larky/src/main/java/com/verygood/security/larky/modules/vgs/cerebro/text/analyzer/dto/TextPIIEntity.java

```diff
@@ -69,16 +72,24 @@ public StarlarkInt end() {
     return end;
   }
 
-  @StarlarkMethod(
-      name = "to_dict",
-      doc = "Dictionary PII representation")
-  public Dict<String, Object> toDict() {
+  @Override
+  @StarlarkMethod(name = PyProtocols.__DICT__, structField = true)
+  public Dict<String, Object> dunderDict() {
+    return TextPIIEntity.dictOf(this.entityType, this.score, this.start, this.end);
+  }
+
+  @Override
+  public boolean isImmutable() {
+    return true;
```

Comment:
`isImmutable()` is what is used to check to see if it is `hashable` and so can be used as a `dict` entry. Is this accurate here?

Reply:
It is true, but it will be safer to put false here, as it is should not be used as key.